### PR TITLE
Task 2  delete todos

### PR DESCRIPTION
**Pull request by rka-wtag**
This PR introduces the functionality of deleting a todo

**Types of Changes**

- Created a delete button alongside the todo creation feature.

[Screencast from 23-11-23 02:26:05 অপরাহ্ণ +06.webm](https://github.com/rka-wtag/todo-app/assets/144114330/b4c0ebe2-9eaf-4cfb-8640-a4c6ae8229b7)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    inputElement.type = "text";
+    inputElement.value = todo.text;
+    inputElement.setAttribute("readonly", "readonly")
+
+    return inputElement;
+}
+
+const handleDeleteTodo = (todo) => {
+    todoList$.innerHTML = null;
+    console.log(todo);
```
Comment (ray-wtag) [inline app.js]: remove please

File: app.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    inputElement.classList.add("text");
+    inputElement.type = "text";
+    inputElement.value = todo.text;
+    inputElement.setAttribute("readonly", "readonly")
+
+    return inputElement;
+}
+
+const handleDeleteTodo = (todo) => {
+    todoList$.innerHTML = null;
```
Comment (ray-wtag) [inline app.js]: ?

File: index.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          >>
+
+          <input id="todo-submit-button" type="submit" value="Submit" />
+        </form>
+      </div>
+    </header>
+
+    <main>
+      <h1>Todo list</h1>
+
```
Comment (ray-wtag) [inline index.html]: Please remove unnecessary spaces

File: index.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          <label for="name">Name:</label>
+
+          <textarea
+            type="text"
+            id="todo-input-box"
+            name="new-task-input"
+            rows="1"
+            required
+          ></textarea
+          >>
```
Comment (ray-wtag) [inline index.html]: ?

File: app.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+
+import {
+    todoInputBox$,
+    todoList$,
```
Comment (ray-wtag) [inline app.js]: why this $ in variable name?

File: app.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
+
+import {
+    todoInputBox$,
```
Comment (ray-wtag) [inline app.js]: ditto

File: app.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+const handleDeleteTodo = (todo) => {
+    todoList$.innerHTML = null;
+    console.log(todo);
+    todos = todos.filter(todoElement => todoElement.id !== todo.id);
+    renderTodos();
+}
+
+const createTodoDeleteButton = (todo) => {
```
Comment (ray-wtag) [inline app.js]: on function naming we don't need to add `todo` everywhere as we already know what the project is. Rather we can say `onCreateDeleteButton`

File: app.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    const todoDeleteButton = document.createElement('button');
```
Comment (ray-wtag) [inline app.js]:
```suggestion
    const deleteButton = document.createElement('button');
```

File: app.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    todoDeleteButton.classList.add('todo-delete-button');
+    todoDeleteButton.innerText = 'Delete';
+
+    todoDeleteButton.addEventListener('click', () => {
+        handleDeleteTodo(todo);
```
Comment (ray-wtag) [inline app.js]:
```suggestion
        handleDelete(todo);
```

File: app.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    })
+    return todoDeleteButton;
+}
+
+const createTodoElement = (todo) => {
```
Comment (ray-wtag) [inline app.js]:
```suggestion
const onCreateElement = (todo) => {
```

File: app.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    const todo$ = document.createElement('li');
```
Comment (ray-wtag) [inline app.js:36]: ditto

File: app.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+import {
+    todoInputBox$,
+    todoList$,
+    todoSubmitButton$
+} from "./domElements.js";
+
+
+let todos = [];
+
+const createInputElement = (todo) => {
```
Comment (ray-wtag) [inline app.js]:
```suggestion
const onCreateInputField = (todo) => {
```

File: index.html [hunk 13]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Document</title>
+  </head>
+  <body>
+    <header>
+      <div>
```
Comment (ray-wtag) [inline index.html]: maybe instead of writing all of these in index.html can't we add a subcomponent?

**Review by ray-wtag — CHANGES_REQUESTED** [review]

File: app.js (hunk 5 above)
Comment (rka-wtag) [reply app.js]: to mark it as a dom element bhai. Should i remove it?

File: app.js (hunk 7 above)
Comment (rka-wtag) [reply app.js]: okay bhai. got it

File: index.html (hunk 13 above)
Comment (rka-wtag) [reply index.html]: bhai can  you please give an example?

File: app.js [hunk 14]
```diff
@@ -50,7 +65,7 @@ const renderTodos = () => {
     todoList$.innerHTML = null;
 
     todos.forEach(todo => {
-        todoList$.appendChild(createTodoElement(todo));
+        todoList$.appendChild(onCreateElement(todo));
```
Comment (hhu-wtag) [inline app.js]: The `on` prefix in a function suggests that, the function is supposed to be called **on** an event. So generally when we attach event listeners to elements such as **buttons** and **inputs**, we prefix the function with **on** keyword. If a function isn't used as event listener, we can omit the **on** prefix and name it normally.

```suggestion
        todoList$.appendChild(createElement(todo));
```

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**